src/lib.rs

```diff
@@ -915,6 +915,18 @@ impl<T> Async<T> {
     }
 }
 
+impl<T> std::convert::AsRef<T> for Async<T> {
```

Comment:
```suggestion
impl<T> AsRef<T> for Async<T> {
```

nit: AsRef/AsMut is included in prelude,